homeassistant/components/swiss_public_transport/config_flow.py

```diff
@@ -30,38 +43,43 @@
 class SwissPublicTransportConfigFlow(ConfigFlow, domain=DOMAIN):
     """Swiss public transport config flow."""
 
-    VERSION = 1
-    MINOR_VERSION = 2
+    VERSION = 2
+    MINOR_VERSION = 1
 
     async def async_step_user(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
         """Async user step to set up the connection."""
         errors: dict[str, str] = {}
         if user_input is not None:
-            await self.async_set_unique_id(
-                f"{user_input[CONF_START]} {user_input[CONF_DESTINATION]}"
-            )
+            unique_id = unique_id_from_config(MappingProxyType(user_input))
```

Comment:
```suggestion
            unique_id = unique_id_from_config(user_input)
```
user_input is a dict